src/utils/analysis-utils.ts

```diff
@@ -670,6 +647,59 @@ export class BaselineLayerResult {
   }
 }
 
+export function getAnalysisTableColumns(
+  analysisResult?: AnalysisResult,
+  withLocalName = false,
+): Column[] {
+  if (!analysisResult || analysisResult instanceof ExposedPopulationResult) {
+    return [];
+  }
+  if ('tableColumns' in analysisResult) {
+    return (analysisResult as PolygonAnalysisResult).tableColumns;
+  }
+  const { statistic } = analysisResult;
+  const baselineLayerTitle = analysisResult.getBaselineLayer().title;
+
+  return [
+    {
+      id: withLocalName ? 'localName' : 'name',
+      label: 'Name',
+    },
+    {
+      id: statistic,
+      label: invert(AggregationOperations)[statistic], // invert maps from computer name to display name.
+      format: value => getRoundedData(value as number),
+    },
+    {
+      id: 'baselineValue',
+      label: baselineLayerTitle,
+      format: (value: number | string) => value.toLocaleString('en-US'),
+    },
+  ];
+}
+
+export function useAnalysisTableColumns(
+  analysisResult?: AnalysisResult,
+): {
+  translatedColumns: Column[];
+  analysisTableColumns: Column[];
+} {
+  const { t, i18n } = useSafeTranslation();
+  return useMemo(() => {
```

Comment:
What was the motivation for using `useMemo`? Makes sense here, but it likely hinders performance than improves it for simple operations like this (mostly used if fed into `useEffect` or other hooks with dependency arrays, or in special cases where the computation is genuinely expensive and slowdowns are noticed)

Reply:
yup if you identified its needed to prevent substantial re-rendering then all good to keep